CharlieBackend.Business/Services/MentorService.cs

```diff
@@ -297,5 +301,25 @@ public async Task<IList<MentorCoursesDto>> GetMentorCoursesByMentorIdAsync(long
 
             return foundCourses;
         }
+
+        /// <summary>
+        /// Return result of checking to role of mentor and it's identity numbers
+        /// </summary>
+        /// <typeparam name="T">Type of data</typeparam>
+        /// <param name="id">Id of user entity</param>
+        /// <param name="result">Result of checking to role and
+        /// equality identity numbers of entities with type of data</param>
+        /// <returns></returns>
+        public Task CheckRoleAndIdMentor<T>(long id, ref Result<T> result)
```

Comment:
Remove the result parameter and return it from the method.
`public Result<T> CheckRoleAndIdMentor<T>(long id)`